docs/2_2_common_mechanisms.adoc

```diff
@@ -1,31 +1,36 @@
 === General Mechanisms [[general-mechanisms]]
 
-This section contains the common interface definitions that allow a C program to invoke the FMU functions.
+==== Requirements for Implementations of the C-API
 
-Note that the following general properties hold for an FMU:
+The following general requirements for implementations of FMUs and importers must be followed:
 
-* FMI functions of one instance do not need to be thread-safe. +
+* FMI functions of one instance are not required to be thread-safe. +
 _[For example, if the functions of one instance of an FMU are accessed from more than one thread;
 the multi-threaded simulation environment that uses the FMU must guarantee that there are no race conditions while invoking the FMI functions.
 The FMU itself does not implement any services to support this.]_
 
-* FMI callback functions are not allowed to call back into the FMU (<<logMessage,`fmi3LogMessageCallback`>>, <<preemption-support,`fmi3LockPreemptionCallback`>>, <<preemption-support,`fmi3UnlockPreemptionCallback`>>), except <<intermediateUpdate,`fmi3IntermediateUpdateCallback`>> and <<fmi3ClockUpdateCallback>> which lead to the well-defined states <<IntermediateUpdateMode>> and <<ClockUpdateMode>>, respectively, defining which function calls back into the FMU are allowed.
+* The following FMI callback functions must not call back into the FMU:
+ ** <<logMessage,`fmi3LogMessageCallback`>>, and
+ ** <<preemption-support,`fmi3LockPreemptionCallback`>>, and
+ ** <<preemption-support,`fmi3UnlockPreemptionCallback`>>).
+
++
+The following callback functions lead to well-defined states and may call FMI functions according to their respective state definitions:
+ ** <<intermediateUpdate,`fmi3IntermediateUpdateCallback`>>, and
+ ** <<fmi3ClockUpdateCallback>>.
 
 * FMI functions must not change global settings which affect other processes/threads.
 An FMI function may change settings of the thread in which it is called (such as floating point control registers), provided these changes are restored before leaving the function or before a callback function is called. +
 _[This property ensures that functions of different FMU instances can be called safely in any order._
 _Additionally, they can be called in parallel provided the functions are called in different processes._
-_If an FMI function changes for example the floating point control word of the CPU, it must restore the previous value before return of the function._
-_For x86 CPUs, the floating point control word is set using the `fldcw` instruction._
```

Comment:
This non-normative text seems to be removed. Should it go the implementers guide?

Reply:
The first sentence is still there, just added commas.
The second is indeed removed as it is a detail for a single platform, when we really say in the example already what to do - the "how for x86" is just distracting.